src/wallet/wallet.cpp

```diff
@@ -2534,8 +2534,17 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend, CWalletTx& wt
                     return false;
                 }
 
-                if (nFeeRet >= nFeeNeeded)
+                if (nFeeRet >= nFeeNeeded) {
+                    // Reduce fee to only the needed amount if we have change output to increase
+                    // Do not do this if fee was paid by payee
+                    if (nFeeRet > nFeeNeeded && nChangePosInOut != -1 && nSubtractFeeFromAmount == 0) {
```

Comment:
Yes my concern is the gap between desired change amounts (CENT/2) and fee overpayments is quite big.  Most of the time even if a fee overpayment was bigger than dust, it wouldn't be so big that you'd actually want to create an output with it.  

I think fixing that should be combined with more robust handling of the minimum output size a wallet would create.  Perhaps then there would be cases where we feel we got unlucky with coin selection and we try again to have less overpayment or more reasonable size change.

The coin selection algorithm is stupid now.  Trying to find an exact match makes no sense in a world where every transaction pays fee.  So I'd rather not continue adding patches and improvements before we try to improve the basic strategy.  I'm happy to give that a shot after 0.14.  And I think we can avoid most of these problems from the beginning...